packages/eas-cli/src/credentials/ios/api/graphql/queries/AppleDeviceQuery.ts

```diff
@@ -41,7 +41,7 @@ const AppleDeviceQuery = {
     );
     const { appleDevices } = data.appleTeam.byAppleTeamIdentifier;
     assert(appleDevices, 'Apple Devices should be defined in this context - enforced by GraphQL');
-    return appleDevices;
+    return appleDevices.filter(device => device) as AppleDevice[];
```

Comment:
We should make the type of the `appleDevices` field stricter in the GraphQL schema defined on the server.

The current type is:
```
[AppleDevice]
```
which means both the array and also each item of the array can be nullable.
`[AppleDevice!]` would be a lot better as it would let us not worry about the individual array items possibly being null (I believe they're already not null, so this type would be more correct). And ideally `[AppleDevice!]!`, which additionally guarantees that the `appleDevices` field is not null, but e.g. `[]` if there are no devices.